pages/data/index.vue

```diff
@@ -271,18 +282,59 @@ export default {
     if (!this.$route.query.type) {
       const firstTabType = compose(propOr('', 'type'), head)(searchTypes)
 
-      this.$router
-        .replace({ query: { type: firstTabType, ...this.$route.query } })
-        .then(() => {
-          this.fetchFromContentful()
-        })
+      this.$router.replace({ query: { type: firstTabType } })
     } else {
-      this.fetchFromContentful()
-      this.fetchFilters()
+      this.fetchResults()
     }
   },
 
   methods: {
+    /**
+     * Figure out which source to fetch results from based on the
+     * type of search
+     */
+    fetchResults: function() {
+      const source = propOr('contentful', 'dataSource', this.searchType)
+
+      const searchSources = {
+        contentful: this.fetchFromContentful,
+        blackfynn: this.fetchFromBlackfynn
+      }
+
+      if (typeof searchSources[source] === 'function') {
+        searchSources[source]()
+      }
+
+      // Fetch filters for the search type
+      this.fetchFilters()
+    },
+
+    /**
+     * Get Search results
+     * This is using fetch from the Blackfynn API
+     */
+    fetchFromBlackfynn: function() {
+      this.isLoadingSearch = true
+
+      const searchType = pathOr('', ['query', 'type'], this.$route)
+      const query = pathOr('', ['query', 'q'], this.$route)
+      const url = `${process.env.discover_api_host}/search/${searchType}s?offset=${this.searchData.skip}&limit=${this.searchData.limit}&organization=SPARC%20Consortium&query=${query}`
```

Comment:
Can we make this a computed property instead?

Reply:
I kept it not a computed value because we need to pass some more conditional logic in to it, which you'll see in #22

Reply:
Sounds good, thanks for the clarification!